test/rewards_test.py

```diff
@@ -0,0 +1,195 @@
+import numpy as np
+from driving_gridworld.rewards import Deterministic_Reward
+from driving_gridworld.rewards import Stochastic_Reward
+from driving_gridworld.rewards import \
+    sample_reward_parameters, \
+    r
+from driving_gridworld.road import Road
+from driving_gridworld.obstacles import Bump
+from driving_gridworld.obstacles import Pedestrian
+from driving_gridworld.car import Car
+from driving_gridworld.actions import ACTIONS, RIGHT, NO_OP, LEFT, UP, DOWN
+import pytest
+
+
+def test_all_entries_in_u_vector_are_greater_than_d_vector():
+    hlr = 4
+    u, C, d, H = sample_reward_parameters(hlr)
+    for i in range(0, hlr + 2):
+        assert u[i] > d[i]
+
+
+def test_all_entries_in_u_vector_are_greater_than_C_matrix_which_are_greater_than_H_matrix(
+):
+    hlr = 4
+    u, C, d, H = sample_reward_parameters(hlr)
+    for i in range(0, hlr + 2):
+        for j in range(0, hlr + 1):
+            assert u[i] > C[i, j] > H[i, j]
+
+
+def test_all_entries_in_d_vector_are_greater_than_H_matrix():
+    hlr = 4
+    u, C, d, H = sample_reward_parameters(hlr)
+    for i in range(0, hlr + 2):
+        for j in range(0, hlr + 1):
+            assert d[i] > H[i, j]
+
+
+def test_hit_a_moving_obstacle_with_higher_speed_gives_lower_reward():
+    hlr = 4
+    u, C, d, H = sample_reward_parameters(hlr)
+    for i in range(0, hlr + 2):
+        for j in range(1, hlr + 1):
+            assert C[i, j - 1] > C[i, j]
+            assert H[i, j - 1] > H[i, j]
+
+
+def test_collision_gives_lower_deterministic_reward_than_no_collision():
```

Comment:
Please spend a bit of time refactoring these tests. I think there should be a way to make them easier to read and perhaps combining some or their set up procedures.